java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java

```diff
@@ -385,10 +397,130 @@ private API getXPAPI(
+          sym = ASTHelpers.getSymbol(identifier);
+        } else if (methodMemberTree.getExpression().getKind() == Kind.MEMBER_SELECT) {
+          // Case 4: Qualified enum name with method call (e.g.
+          // "TestExperimentName.STALE_FLAG.getKey()")
+          MemberSelectTree memberTreeExpression =
+              (MemberSelectTree) methodMemberTree.getExpression();
+          sym = ASTHelpers.getSymbol(memberTreeExpression);
+        } else {
+          return false;
+        }
+        break;
+      default:
+        return false;
+    }
+
+    if (sym.getKind() != ElementKind.ENUM_CONSTANT) {
+      return false;
+    }
+
+    Symbol.ClassSymbol enclosingClass = ASTHelpers.enclosingClass(sym);
+
+    return isEnumConstantMatchingFlagName(sym.getSimpleName().toString(), enclosingClass, state);
+  }
+
+  /**
+   * Finds whether the current enum constant contains the flag string in its constructor arguments.
+   * Must also have a valid configuration record corresponding to this enum in order to return true.
+   *
+   * @param enumName Unqualified enum constant name (e.g. STALE_FLAG)
+   * @param classSymbol Enum class symbol for the corresponding constant
+   * @param state Used to find enum class in AST
+   */
+  public boolean isEnumConstantMatchingFlagName(
+      String enumName, Symbol.ClassSymbol classSymbol, VisitorState state) {
+
+    // Skip any enums that have no configuration
+    ImmutableCollection<PiranhaEnumRecord> enumRecords =
+        this.config.getEnumRecordsForName(classSymbol.getSimpleName().toString());
+    if (enumRecords.isEmpty()) {
+      return false;
+    }
+
+    // Check cached matches
+    EnumWithClassSymbol enumWithClassSymbol = new EnumWithClassSymbol(enumName, classSymbol);
+    if (enumsMatchingConstructorArgsCache.containsKey(enumWithClassSymbol)) {
+      return enumsMatchingConstructorArgsCache.get(enumWithClassSymbol);
+    }
+
+    // TODO: Find constructor arguments purely through the symbol table; without the need for state
+    //       Then, we will be able remove the following exception
+    ClassTree enumClassTree = ASTHelpers.findClass(classSymbol, state);
+    if (enumClassTree == null) {
+      throw new PiranhaRuntimeException(
+          "Detected enum constant of class "
+              + classSymbol.className()
+              + ", which is mentioned by"
+              + " Piranha's configuration as part of enumProperties. However, enum definition source"
+              + " is not available when looking at this usage (this can happen when cleaning up flags"
+              + " across build targets)."
+              + "\n\nIf you are trying to use Piranha to clean up enum flags based on the string"
+              + " argument to their constructor (e.g. using enumProperties), the current"
+              + " implementation will fail to match flag usages on a separate target as that"
+              + " containing the enum source, resulting in partial clean up.");
```

Comment:
👍  (There is a chance that we discover that in practice erroring out here is too strict and we should print a warning, but I definitely prefer to start with the most "fail-fast" possible solution until we know for sure how often this case shows up or if there is any better way to handle it! :) )

Reply:
I'm generally in favor of failing fast 😁 Hiding errors tends to cause problems later...